linkerd/app/integration/src/tests/transparency.rs

```diff
@@ -709,6 +709,96 @@ macro_rules! http1_tests {
             proxy.join_servers().await;
         }
 
+        /// Tests that the proxy drops headers that can't be used in CONNECT responses.
+        ///
+        /// Exercises https://tools.ietf.org/html/rfc7231#section-4.3.6:
+        ///
+        /// A client MUST ignore any Content-Length or Transfer-Encoding header
+        /// fields received in a successful response to CONNECT.
+        #[tokio::test]
+        async fn http11_connect_headers_stripped() {
+            let _trace = trace_init();
+
+            // To simplify things for this test, we just use the test TCP
+            // client and server to do an HTTP CONNECT.
+            //
+            // We don't *actually* perfom a new connect to requested host,
+            // but client doesn't need to know that for our tests.
+
+            let connect_req = b"\
+                               CONNECT transparency.test.svc.cluster.local HTTP/1.1\r\n\
+                               Host: transparency.test.svc.cluster.local\r\n\
+                               \r\n\
+                               ";
+            let connect_res = b"\
+                               HTTP/1.1 200 OK\r\n\
+                               content-length: 0\r\n\
+                               \r\n\
+                               ";
+
+            let tunneled_req = b"{send}: hi all\n";
+            let tunneled_res = b"{recv}: welcome!\n";
+
+            let srv = server::tcp()
+                .accept_fut(move |mut sock| {
+                    async move {
+                        // Read connect_req...
+                        let mut vec = vec![0; 512];
+                        let n = sock.read(&mut vec).await?;
+                        let head = s(&vec[..n]);
+                        assert_contains!(
+                            head,
+                            "CONNECT transparency.test.svc.cluster.local HTTP/1.1\r\n"
+                        );
+
+                        // Write connect_res back...
+                        sock.write_all(&connect_res[..]).await?;
+
+                        // Read the message after tunneling...
+                        let mut vec = vec![0; 512];
+                        let n = sock.read(&mut vec).await?;
+                        assert_eq!(s(&vec[..n]), s(&tunneled_req[..]));
+
+                        // Some processing... and then write back tunneled res...
+                        sock.write_all(&tunneled_res[..]).await
+                    }
+                    .map(|res: std::io::Result<()>| match res {
+                        Ok(()) => {}
+                        Err(e) => panic!("tcp server error: {}", e),
+                    })
```

Comment:
tiny nit, TIOLI: i believe this could avoid matching the `Ok(())` case if it used the [`unwrap_or_else`combinator](https://docs.rs/futures/latest/futures/future/trait.TryFutureExt.html#method.unwrap_or_else) rather than `map`:
```suggestion
                    .unwrap_or_else(|e| panic!("tcp server error: {}", e))
```

Reply:
this is copy/paste from above. would love help cleaning up the tests in a separate PR